### ISSUE TITLE
@subcommand Sanitizing

### ISSUE THREAD

**Issue by joshstoik1** [wp-cli/wp-cli#4159]
When a subcommand or alias is specified using the function documentation markup (e.g. `@subcommand xyz` or `@alias xyz`), values containing a number are terminated wherever that number occurs.

For example, I have a class that is registered as a command, and within that class a public function called `revoke_md5_passwords()`. But I want users to access it via "revoke-md5-passwords", so added `@subcommand revoke-md5-passwords` to the function docs. The subcommand that ends up being registered is just "revoke-md".

The subcommand registration does, however, work fine if explicitly calling
`WP_CLI::add_command(
	'foo bar revoke-md5-passwords',
	$functionvar
);`

Long story short, it seems different command name sanitizing is happening depending on how that value makes its way into the system. :)

**Comment by gitlost** [comment]
Yes, the DocParser https://github.com/gitlost/wp-cli/blob/master/php/WP_CLI/DocParser.php#L82 doesn't allow numbers (or uppercase or dot) while the Configurator https://github.com/gitlost/wp-cli/blob/master/php/WP_CLI/Configurator.php#L37 does.
